src/integrations/pulsar/consumer.cpp

```diff
@@ -274,11 +286,13 @@ void Consumer::StartConsuming() {
       spdlog::info("Pulsar consumer {} finished processing", info_.consumer_name);
     }
     is_running_.store(false);
+    remaining_nof_batches_to_read_.reset();
```

Comment:
As for the Kafka consumer this can give data race issue. This should be removed.